baselines/dom.generated.d.ts

```diff
@@ -6533,36 +6533,27 @@ declare var HTMLCanvasElement: {
 };
 
 /** A generic collection (array-like object similar to arguments) of elements (in document order) and offers methods and properties for selecting from the list. */
-interface HTMLCollectionBase {
+interface HTMLCollection<E extends Element = Element, N = never> {
     /**
      * Sets or retrieves the number of objects in a collection.
      */
     readonly length: number;
     /**
      * Retrieves an object from various collections.
      */
-    item(index: number): Element | null;
-    [index: number]: Element;
-}
-
-interface HTMLCollection extends HTMLCollectionBase {
+    item(index: number): E | null;
     /**
      * Retrieves a select object or an object from an options collection.
      */
-    namedItem(name: string): Element | null;
+    namedItem(name: string): E | N | null;
+    [index: number]: E;
 }
 
 declare var HTMLCollection: {
     prototype: HTMLCollection;
     new(): HTMLCollection;
 };
 
-interface HTMLCollectionOf<T extends Element> extends HTMLCollectionBase {
```

Comment:
I and probably other people have used it in own code (API which should usable with `getElementsByTagNameNS` output for example.
So this interface should probably survive with a `@deprecated` tag.

Reply:
The API name also is consistent with `NodeListOf`.

Reply:
Added a deprecated `HTMLCollectionOf` alias for backwards compatibility. @saschanaz should I maybe do the same for `NodeListOf` then for consistency?

Reply:
Ping @saschanaz.

Reply:
I don't have a strong opinion here, maybe @sandersn ?